pkg/processor/sourceprocessor/source_category_filler_test.go

```diff
@@ -62,3 +62,73 @@ func TestFillWithAnnotations(t *testing.T) {
 
 	assertAttribute(t, attrs, "_sourceCategory", "annoPrefix:sc#from#annot#ns#1#123asd")
 }
+
+func TestFillWithContainerAnnotations(t *testing.T) {
+	t.Run("container annotations are disabled by default", func(t *testing.T) {
+		cfg := createDefaultConfig().(*Config)
+		cfg.SourceCategory = "my-source-category"
+
+		attrs := pdata.NewAttributeMap()
+		attrs.InsertString("k8s.pod.annotation.sumologic.com/container-name-1.sourceCategory", "first_source-category")
+		attrs.InsertString("k8s.pod.annotation.sumologic.com/container-name-2.sourceCategory", "another/source-category")
+		attrs.InsertString("k8s.container.name", "container-name-1")
+
+		filler := newSourceCategoryFiller("_sourceCategory", cfg)
+		filler.fill(&attrs)
+
+		assertAttribute(t, attrs, "_sourceCategory", "kubernetes/my/source/category")
+	})
+
+	t.Run("source category for container-name-1", func(t *testing.T) {
+		cfg := createDefaultConfig().(*Config)
+		cfg.SourceCategory = "my-source-category"
+		cfg.ContainerAnnotations.Enabled = true
+
+		attrs := pdata.NewAttributeMap()
+		attrs.InsertString("k8s.pod.annotation.sumologic.com/container-name-1.sourceCategory", "first_source-category")
+		attrs.InsertString("k8s.pod.annotation.sumologic.com/container-name-2.sourceCategory", "another/source-category")
+		attrs.InsertString("k8s.container.name", "container-name-1")
+
+		filler := newSourceCategoryFiller("_sourceCategory", cfg)
```

Comment:
One nit: `newSourceCategoryFiller` seems to be configurable but is always called with `"_sourceCategory"` or `sourceCategoryKey` as the first param + the names says it all.

I believe we can drop this param and just hardcode it, no?

Reply:
in addition we can hardcode all `_source` headers as we treat it special way in exporter

Reply:
Removed `attributeName` parameter in https://github.com/SumoLogic/sumologic-otel-collector/pull/260/commits/375315674010f0e344d265c7ff5d0cbb781a4ab1